code/game/turfs/open.dm

```diff
@@ -123,10 +123,12 @@
 			cleanup(H)
 
 		else if(isxeno(C))
-			if(!isxenoboiler(C))
-				beachwater_slowdown = 1.3
-			else
+			if(isxenoboiler(C))
 				beachwater_slowdown = -0.5
+			else if(isxenowwarlock(C))
+				beachwater_slowdown = 0
+			else
+				beachwater_slowdown = 1.3
```

Comment:
I discovered that boilers actually go FASTER on water a while ago. While that seems slightly insane I thought warlock not getting slowed down since it's floating above the water makes sense.